test/utils.py

```diff
@@ -0,0 +1,42 @@
+from shlex import split
+from subprocess import run
+
+
+def determine_scope(
+    fixture_name,
+    config
+) -> str:
+    """
+    Sets the scope of certain fixtures.
+
+    Parameters
+    ----------
+    fixture_name
+        The name of the fixture.
+    config
+        A pytest internal object.
+
+    Returns
+    -------
+    str
+        The scope name.
+    """
+    if config.option.numprocesses in ["auto", "logical"]:
+        return "function"
+    return "session"
```

Comment:
The first line is checking for pytest-xdist - the "-n" option in that package is recorded as numprocesses. I think I intended to modify this further but it has turned out that these tests start to get buggy if everything isn't just scoped at "function".

In my other branch I just have it always returning "function" but have left it in since I'm still not sure if the bugs at function scope are caused by other problems, and the tests would run faster if they had a wider scope like "session."